services/processor/native/repository/Committee/index.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 the orbs-network-go authors
+// This file is part of the orbs-network-go library in the Orbs project.
+//
+// This source code is licensed under the MIT license found in the LICENSE file in the root directory of this source tree.
+// The above notice should be included in all copies or substantial portions of the software.
+
+package committee_systemcontract
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/orbs-network/orbs-contract-sdk/go/sdk/v1"
+	"github.com/orbs-network/orbs-contract-sdk/go/sdk/v1/service"
+	"github.com/orbs-network/orbs-contract-sdk/go/sdk/v1/state"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/services/processor/native/repository/_Elections"
+)
+
+// helpers for avoiding reliance on strings throughout the system
+const CONTRACT_NAME = "_Committee"
+const METHOD_GET_ORDERED_COMMITTEE = "getOrderedCommittee"
+const METHOD_UPDATE_REPUTATION = "updateReputation"
+
+var PUBLIC = sdk.Export(getOrderedCommittee, updateReputation)
+var SYSTEM = sdk.Export(_init)
+
+const ToleranceLevel = uint32(4)
+const ReputationBottomCap = uint32(10)
+
+func _init() {
+}
+
+func _formatReputation(addr []byte) []byte {
+	return []byte(fmt.Sprintf("Validator_%s_Rep", hex.EncodeToString(addr)))
+}
+
+func _getReputation(addr []byte) uint32 {
+	return state.ReadUint32(_formatReputation(addr))
+}
+
+func _degradeReputation(addr []byte) {
+	currRep := _getReputation(addr)
+	if currRep < ReputationBottomCap {
```

Comment:
for auditability reasons degrade reputation as much as needed - on query - it will retrieve the correct (not the capped on).  On compute of reputation use the bounds (cap above and below)

Reply:
doesn't that make some of the calculation opaque ? since if you audit you need to know the caps ?